File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 3/31/2026, 9:40:37 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxx = -10000
        curr = 0
        for i in range(len(nums)):
            curr = curr + nums[i]
            if (maxx<curr):
                maxx = curr
            if (curr<0):
                curr = 0 
        return maxx

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 3/31/2026, 9:40:45 PM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
    int maxsum = nums[0];
    int currsum = nums[0];

    for (int i = 1; i < nums.size(); i++) {
        currsum = max(nums[i], currsum + nums[i]);
        maxsum = max(maxsum, currsum);
    }

    return maxsum;
}
};